lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -608,20 +667,51 @@
         return toReturn;
     }());
 
+    // mappedDefaultLocale will get the default locale and update any deprecated
+    // collation/sort order values it may use
+    let __mappedDefaultLocale = undefined;
     var mappedDefaultLocale = function () {
-        var parts = platform.builtInRegexMatch(GetDefaultLocale(), /([^_]*)_?(.+)?/);
-        var locale = parts[1];
-        var collation = parts[2];
-        var availableBcpTags = reverseLocaleAcceptingCollationValues[locale];
+        if (__mappedDefaultLocale) {
+            return __mappedDefaultLocale;
+        }
 
-        if (collation === undefined || availableBcpTags === undefined) { return locale; }
+        let locale = undefined;
+        let collation = undefined;
+        if (isPlatformUsingICU) {
+            // ICU's getDefaultLocale() will return a valid BCP-47/RFC 5646 langtag
+            locale = GetDefaultLocale();
+            const match = platform.builtInRegexMatch(locale, /-u(-[^\-][^\-]?-[^\-]+)*-co-([^\-]+).*/);
+            if (match) {
+                collation = match[2];
+                locale = callInstanceFunc(StringInstanceReplace, locale, `-co-${match[2]}`, "");
```

Comment:
>match[2] [](start = 80, length = 8)

nit (TODO): you just set collation = match[2], so use ${collation} instead

Reply:
Here, do we intentionally still replace `-co-value` instead of stripping all subtags? (I'm guessing this mapping possibly involves other subtags?) 
nit (TODO): (Come to think of it `reverseLocaleAcceptingCollationValues` needs a comment or example explaining what the resulting object looks like, because I hate having to use a debugger every time I want to answer that question.)

---
In reply to: [141544228](https://github.com/Microsoft/ChakraCore/pull/3820#discussion_r141544228) [](ancestors = 141544228)

Reply:
As far as I know (and as far as its variable name suggests), `localesAcceptingCollationValues` is only for collation values. As for whether the other extensions need to be stripped, as of _right now_, this function is only used as the `defaultLocaleFunc` for `resolveLocales` when initializing the Collator (initializing NumberFormat and DateTimeFormat uses `strippedDefaultLocale` instead). That internally calls `platform.getExtensions`, which falls back to `getExtensionSubtags(locale)`, which is basically just 

```js
locale
  .match(LANG_TAG_EXT_RE)[0] // get extensions on the locale string
  .split('-') // split them into an array
  .filter(x => !!x) // remove empty elements
```

So, should we strip all subtags? No, probably not. Also, thinking about it, we probably shouldn't be stripping all subtags for NumberFormat or DateTimeFormat either. Id imagine the original reason for the different functions was that Collator needed the system collation value, so it went through the process of stripping "_collation" and mapping and re-formatting it, while NumberFormat and DateTimeFormat didn't need the collation, so they could blindly throw away the "_collation" part of the string, because they knew they could get -ca and -nu formats later from platform.getExtensions. In ICU, we cant do that right now and possibly don't need to do that ever. Also, since we are caching the results now, the perf cost of calculating the mapped collation value (_if we even need to_, re: the comment below) is 0 for the Number and Date case.

TL;DR: Stripping the locale for ICU is at best useless and at worst harmful, so I think we should actually ditch the ICU implementation of strippedDefaultLocale regardless